fastlane/lib/fastlane/actions/git_commit.rb

```diff
@@ -10,8 +10,17 @@ def self.run(params)
 
         skip_git_hooks = params[:skip_git_hooks] ? '--no-verify' : ''
 
-        result = Actions.sh("git commit -m #{params[:message].shellescape} #{paths} #{skip_git_hooks}".strip)
-        UI.success("Successfully committed \"#{params[:path]}\" 💾.")
+        command = "git commit -m #{params[:message].shellescape} #{paths} #{skip_git_hooks}".strip
+        nothing_to_commit = false
+
+        if params[:allow_nothing_to_commit]
+          repo_clean = Actions.sh("git status --porcelain").empty?
+          UI.success("Nothing to commit, working tree clean ✅.") if repo_clean
+          return if repo_clean
+        end
+
+        result = Actions.sh(command)
+        UI.success("Successfully committed \"#{params[:path]}\" 💾.") unless nothing_to_commit
```

Comment:
Can remove the `unless nothing_to_commit` since the clean repo is early returning 💣